### ISSUE TITLE
REST API POST-Request not working

### ISSUE THREAD

**Issue by Ghostw4lk** [esl/MongooseIM#1719]
MongooseIM version: (put the version)
2.1.1

Installed from: (source or pkg)
Source

Erlang/OTP version: (put the version)
20.0

Describe the issue:
I want to create users through the REST API with JavaScript, but the response is always "undefined".

The configuration of the mongoose_api_admin is the following:
```xml
  %% MongooseIM HTTP API it's important to start it on localhost
  %% or some private interface only (not accessible from the outside)
  %% At least start it on different port which will be hidden behind firewall

  { {8088, "0.0.0.0"} , ejabberd_cowboy, [
      {num_acceptors, 10},
      {transport_options, [{max_connections, 1024}]},
      {modules, [
          {"localhost", "/api", mongoose_api_admin, []}
      ]}
  ]},
```

This is my JavaScript code:
```xml
var url = "http://<<MyServerIP>>:8088/api/users/localhost";

var asyncRequest = new XMLHttpRequest();

asyncRequest.open("POST", url, true);
asyncRequest.setRequestHeader('Content-type','application/json');

asyncRequest.onreadystatechange = function () {
	//var users = JSON.parse(asyncRequest.responseText);
	if (asyncRequest.readyState == 4 && asyncRequest.status == "201") {
		console.out("User registered");
	} else {
		console.error("Error: " + asyncRequest.responsetext);
	}
}
asyncRequest.send(JSON.stringify({username:"testuser", password:"12345"}));
```

The JavaScript code is executed from my computer. Thats the reason why I changed the IP from 127.0,0.1 to "0.0.0.0", to access the API from outside for the development.

Any idea why the response is: undefined?
I used the documentation to create a 'Register user' -request.

Thats the cUrl command: 
```xml
curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' -d '{ \ 
   "username": "testuser", \ 
   "password": "12345" \ 
 }' 
```

Any help is appreciated.

**Comment by Ghostw4lk** [comment]
I have read in other issues that I have to change the "localhost to "_" and the IP "127.0.0.1" to my "Server IP", but this does not work.

My request URL is now: "http://<<Server-IP>>:8088/api/users"

**Comment by Ghostw4lk** [comment]
Okay I got it.
The error was that the curl command was splitted with "\" and the server could handle this request.
So I removed the "\" from the command and the request works well.

The second problem was in the JavaScript code.
I don´t know what the difference is but I have to make the request with this URL: "http://localhost:8088/api/users/localhost", before it was "http://127.0.0.1:8088/api/users/localhost".
